### PR TITLE
Feature/api

### PR DESCRIPTION
**Pull request by recroot89**

### CODE REVIEW

File: app/controllers/api/v1/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class Api::V1::TasksController < Api::V1::ApplicationController
+  def index
+    q_params = params[:q] || { s: 'id asc' }
+
+    tasks = Task.all
```
Comment (pn-y) [inline app/controllers/api/v1/tasks_controller.rb]: .all is useless here

File: app/controllers/api/v1/tasks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    respond_with(task)
+  end
+
+  def create
+    task = current_user.my_tasks.new(task_params)
+
+    if task.save
+      respond_with(task, location: nil)
+    else
+      render(json: { errors: task.errors }, status: :unprocessable_entity)
```
Comment (pn-y) [inline app/controllers/api/v1/tasks_controller.rb]: respond_with know how to manage cases when entity wasn't saved

File: app/serializers/task_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class TaskSerializer < ActiveModel::Serializer
+  attributes :id, :name, :description, :state, :expired_at
+  belongs_to :author
+  belongs_to :assignee
+
+  def id
+    object.id.to_s
```
Comment (pn-y) [inline app/serializers/task_serializer.rb]: any reason to have string id?

**Review by pn-y — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/tasks_controller.rb (hunk 1 above)
Comment (recroot89) [reply app/controllers/api/v1/tasks_controller.rb]: thanks! Didn't notice

File: app/serializers/task_serializer.rb (hunk 3 above)
Comment (recroot89) [reply app/serializers/task_serializer.rb]: it works without stringify. may be it is useful when we want to be sure the result is always a string?

**Comment by pn-y** [comment]
You (almost) always will have id as int

File: app/controllers/api/v1/tasks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    respond_with(task)
+  end
+
+  def create
+    task = current_user.my_tasks.new(task_params)
+
+    if task.save
+      respond_with(task, location: nil)
+    else
+      respond_with(task.errors, status: :unprocessable_entity)
```
Comment (pn-y) [inline app/controllers/api/v1/tasks_controller.rb]: What do you think will happen if you just write
```
task.save
respond_with task
```

Comment (recroot89) [reply app/controllers/api/v1/tasks_controller.rb]: thanks!
https://www.rubydoc.info/github/plataformatec/responders/ActionController%2FRespondWith:respond_with
http://blog.plataformatec.com.br/2009/08/embracing-rest-with-mind-body-and-soul/
It helped to understand. I hope now it is correct.

File: app/controllers/api/v1/tasks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    respond_with(task, location: nil)
+  end
+
+  def update
+    task = Task.find(params[:id])
+
+    if task.update(task_params)
+      render(json: task)
+    else
+      render(json: { errors: task.errors }, status: :unprocessable_entity)
```
Comment (pn-y) [inline app/controllers/api/v1/tasks_controller.rb]: Use respond_with

File: app/controllers/api/v1/tasks_controller.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    end
+  end
+
+  def destroy
+    task = Task.find(params[:id])
+    if task.destroy
+      head(:ok)
+    else
+      render(json: { errors: task.errors }, status: :unprocessable_entity)
```
Comment (pn-y) [inline app/controllers/api/v1/tasks_controller.rb]: Use respond_with

File: app/serializers/task_serializer.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class TaskSerializer < ActiveModel::Serializer
+  attributes :id, :name, :description, :state, :expired_at
+  belongs_to :author
+  belongs_to :assignee
+
+  def id
```
Comment (pn-y) [inline app/serializers/task_serializer.rb]: Useless method

**Review by pn-y — CHANGES_REQUESTED** [review]

**Review by pn-y — APPROVED** [review]

**Pull request merged**